packages/jumpstarter-driver-snmp/jumpstarter_driver_snmp/driver.py

```diff
@@ -0,0 +1,166 @@
+            "my-target",
+            udp.DOMAIN_NAME,
+            (self.ip_address, self.port),
+            "my-creds"
+        )
+
+        return snmp_engine
+
+    @classmethod
+    def client(cls) -> str:
+        return "jumpstarter_driver_snmp.client.SNMPServerClient"
+
+    def _snmp_set(self, state: PowerState):
+        result = {"success": False, "error": None}
+
+        def callback(snmpEngine, sendRequestHandle, errorIndication,
+                    errorStatus, errorIndex, varBinds, cbCtx):
+            self.logger.info(f"Callback {errorIndication} {errorStatus} {errorIndex} {varBinds}")
+            if errorIndication:
+                self.logger.info(f"SNMP error: {errorIndication}")
+                result["error"] = f"SNMP error: {errorIndication}"
+            elif errorStatus:
+                self.logger.info(f"SNMP status: {errorStatus}")
+                result["error"] = (
+                    f"SNMP error: {errorStatus.prettyPrint()} at "
+                    f"{varBinds[int(errorIndex) - 1][0] if errorIndex else '?'}"
+                )
+            else:
+                result["success"] = True
+                for oid, val in varBinds:
+                    self.logger.debug(f"{oid.prettyPrint()} = {val.prettyPrint()}")
+            self.logger.info(f"SNMP set result: {result}")
+
+        try:
+            self.logger.info(f"Sending power {state.name} command to {self.host}")
+
+            snmp_engine = self._setup_snmp()
+
+            cmdgen.SetCommandGenerator().send_varbinds(
+                snmp_engine,
+                "my-target",
+                None,
+                "",
+                [(self.full_oid, rfc1902.Integer(state.value))],
+                callback,
+            )
+
+            snmp_engine.open_dispatcher(self.timeout)
+            snmp_engine.close_dispatcher()
+
+            if not result["success"]:
+                raise SNMPError(result["error"])
+
+            return f"Power {state.name} command sent successfully"
+
+        except Exception as e:
+            error_msg = f"SNMP set failed: {str(e)}"
+            self.logger.error(error_msg)
+            raise SNMPError(error_msg) from e
+
+    @export
+    def power_on(self):
+        """Turn power on"""
+        return self._snmp_set(PowerState.ON)
+
+    @export
+    def power_off(self):
+        """Turn power off"""
+        return self._snmp_set(PowerState.OFF)
+
+    @export
+    def power_cycle(self):
+        """Power cycle the device"""
+        try:
+            self.logger.info("Starting power cycle sequence")
+            self.power_off()
+            self.logger.info(f"Waiting {self.quiescent_period} seconds...")
+            time.sleep(self.quiescent_period)
+            self.power_on()
+            return "Power cycle completed successfully"
+        except Exception as e:
+            error_msg = f"Power cycle failed: {str(e)}"
+            self.logger.error(error_msg)
+            raise SNMPError(error_msg) from e
```

Comment:
```suggestion
    def power_cycle(self):
        """Power cycle the device"""
        try:
            self.logger.info("Starting power cycle sequence")
            self.power_off()
            self.logger.info(f"Waiting {self.quiescent_period} seconds...")
            time.sleep(self.quiescent_period)
            self.power_on()
            return "Power cycle completed successfully"
        except Exception as e:
            error_msg = f"Power cycle failed: {str(e)}"
            self.logger.error(error_msg)
            raise SNMPError(error_msg) from e
```

Let's move this as a client side implementation to the jumpstarter_driver_power.client.PowerClient.